### ISSUE TITLE
Torch auto-configurator, optimize usage

### ISSUE THREAD

**Issue by shindavid** [shindavid/AlphaZeroArcade#18]
Using DshinScratch.cpp as an example, create a process that allows the user to determine optimal configuration specific to their local hardware. We want a nnet batch size that will permit maximum evaluation throughput. This will be game-specific.

Later this can be part of a general auto-configurator that determines other parameters like the number of search threads to use.

@Rediness remarked that the use of `.detach()` was probably suboptimal. Get to the bottom of this.
